CMakeLists.txt

```diff
@@ -61,40 +49,136 @@ endif()
 file(READ VERSION H3_VERSION LIMIT_COUNT 1)
 # Clean any newlines
 string(REPLACE "\n" "" H3_VERSION "${H3_VERSION}")
+set(H3_SOVERSION 1)
 
-find_library(M_LIB m)
+project(h3 VERSION ${H3_VERSION})
 
-include_directories(src/h3lib/include)
+include(CMakeDependentOption)
+include(CTest)
 
-file(GLOB_RECURSE ALL_SOURCE_FILES src/*.c src/*.h)
-file(GLOB_RECURSE EXAMPLE_SOURCE_FILES examples/*.c)
+find_library(M_LIB m)
+
+set(LIB_SOURCE_FILES
```

Comment:
I think it helps make sure you actually build the files in question. Otherwise, make might not have info on a new file. And sure, my vim is just set to two spaces.